quickwit/quickwit-metastore/src/metastore/file_backed_metastore/mod.rs

```diff
@@ -198,7 +205,20 @@ impl FileBackedMetastore {
         }
     }
 
-    async fn read<T, F>(&self, index_id: &str, view: F) -> MetastoreResult<T>
+    async fn read<T, F>(&self, index_uid: IndexUid, view: F) -> MetastoreResult<T>
+    where F: FnOnce(&FileBackedIndex) -> MetastoreResult<T> {
+        let index_id = index_uid.index_id();
+        let locked_index = self.get_locked_index(index_id).await?;
+        if locked_index.index_uid() == index_uid {
+            view(&locked_index)
+        } else {
+            Err(MetastoreError::IndexDoesNotExist {
+                index_id: index_id.to_string(),
+            })
+        }
+    }
+
+    async fn read_any<T, F>(&self, index_id: &str, view: F) -> MetastoreResult<T>
```

Comment:
`ready_any` is used once, we could inline it and then have `fn get_locked_index(index_uid: IndexUid)` instead.

Reply:
nvm.